### ISSUE TITLE
Native engine can't load on Linux (hardcoded llama.dll/ggml.dll), plus two calibration/default notes

### ISSUE THREAD

**Issue by MrJev** [sabeel111/OpenSourceJev#1]
Hi — I reviewed OpenSourceJev for a write-up on mrjev.com, at `3c41fba`. I ran it in `python:3.11-slim` in Docker on four CPU cores against llama.cpp `b11065` and `Qwen2.5-0.5B-Instruct-Q8_0`, deliberately not your Windows/CUDA setup, because the README lists Linux x86_64 as supported. Three things, one of them a three-line fix.

The design deserves saying first: tokenising each allowed answer and reading `llama_get_logits_ith` at the candidate positions is the right way to do this, and the trace you return — probability, logprob *and* token count per candidate — is the most informative I've seen in this group. When a decision surprises you, that trace tells you why.

**1. The native engine can't load on Linux.**

`_library_path()` defaults to `runtime/llama.cpp/bin/llama.dll`, and `_load_library()` then loads `ggml.dll` from the same directory unconditionally, so `JEV_LLAMA_DLL` pointing at the `.so` doesn't help either:

```
$ JEV_LLAMA_RUNTIME=/lc python -c "from app.llama_ctypes import native_status; print(native_status())"
{'available': False, 'library': '/lc/llama.dll', 'error': 'llama.dll not found'}

$ JEV_LLAMA_DLL=/lc/libllama.so python -c "…"
{'available': False, 'error': 'Could not load the prebuilt llama.cpp runtime:
                               /lc/ggml.dll: cannot open shared object file'}
```

Two symlinks (`llama.dll -> libllama.so`, `ggml.dll -> libggml.so`) plus `apt install libgomp1` and everything works — your struct layouts match a current llama.cpp build fine, and a three-option choice came back at 0.930/0.064/0.006 with the logprobs attached, median 512 ms per step on four CPU cores. So the only thing standing between Linux users and a working install is the file name. Something like:

```python
def _default_names():
    if os.name == "nt":
        return "llama.dll", "ggml.dll"
    if sys.platform == "darwin":
        return "libllama.dylib", "libggml.dylib"
    return "libllama.so", "libggml.so"
```

used in `_library_path()` and for the backend load, with the existing env overrides still winning. Happy to send that as a PR if you'd like it.

**2. The BoolQ temperature is applied to whatever GGUF is loaded.**

`jev_calibration.json` is explicit that it was fitted on `Qwen/Qwen3-1.7B`, but `_noul_temperature()` returns 9.4705 for any model, and `sigmoid((logit_true - logit_false) / 9.4705)` compresses the output into a narrow band around 0.5 by construction. On a different model (Qwen2.5-0.5B-Q8):

| question | shipped T=9.4705 | T=1 |
| --- | ---: | ---: |
| "Does this request convey urgency?" (clearly yes) | 0.518 | 0.668 |
| "Is this message about domestic cats?" (clearly no) | 0.412 | 0.033 |

The boolean is still right — the transform is monotone — but 0.412 on a question whose answer is plainly no is a number people will threshold on. A one-line warning when the loaded model path doesn't match `jev_calibration.json`'s `model` field would make it obvious that the calibration has to be refitted, and would cost nothing for users who are on Qwen3-1.7B as intended.

**3. `mode` defaults to `"mock"`, and the mock claims a model.**

`RunRequest.mode` defaults to `"mock"`, so a request that follows the quickstart but omits the field gets the keyword heuristic. That's a defensible default for a demo, and `meta.mode` does say `mock`, which is most of what's needed. The part I'd change is the line next to it:

```json
{"outputs": {"urgent": false, "category": "technical"},
 "meta": {"mode": "mock", "model": "qwen3:1.7b"},
 "trace": [{"id": "urgent", "value": false, "confidence": 0.76}]}
```

Nothing was loaded, but `meta.model` names a model. (The input there was "The checkout is failing for every customer and card authorizations are failing", which the mock rated not-urgent at 0.76 because "failing" isn't in its word list — fine for a mock, less fine when the response looks like a real one.) Reporting `"model": null` in mock mode, or `"model": "(mock, no model loaded)"`, would remove the ambiguity.

**Two smaller notes.**

`_length_penalty_alpha()` returns 0.0 by default, so the length normalisation the README lists as a key feature is off unless `JEV_LENGTH_PENALTY_ALPHA` is set. I tried to make it bite — relabelling the winning option from `billing` (1 token) to `billing department escalation team` (4 tokens) — and it still won, 0.921 vs 0.930, so this looks like documentation rather than a bug.

And `llama_version` is bound but never checked against anything. The struct mirror in `llama_ctypes.py` is copied from one build of llama.cpp; a build with a different layout will read the wrong fields rather than fail. A recorded "known-good" build number in the README, or a soft version check at load, would save someone a confusing afternoon.

For what it's worth: the 16 tests pass, and there's no `.github` directory to run them. `pytest` on push would be cheap here.

**Comment by sabeel111** [comment]
Thanks for doing such a thorough cross-platform test  this is extremely useful. Honestly, to tell you the truth i didnt get enough time to sit on the project, i did a quick prototyping on my windows system and just pushed the code. yes the BoolQ temperature is should only be applied to the model it was calibaerated for. Current caliberation is only for Qwen 3 1.7B Q8 model. The mock mode is not supposed to be there, i forgot to remove it before pushing. 

I'll fix the loader/model metadata and add an explicit calibration-model check. The length-penalty default and llama.cpp version check are also good catches I'll clarify/fix those as well.

The readme is not correct i will fix that as well. 

And thanks for offering the PR for the Linux loader. Happy to review it if you'd like to send it.
